proto/minder/v1/minder.proto

```diff
@@ -2038,6 +2051,19 @@ message AssignRoleResponse {
     Invitation invitation = 2;
 }
 
+message UpdateRoleRequest {
+    // context is the context in which the role assignment is evaluated.
+    Context context = 1;
+    // role_assignments are the role assignments to be updated.
+    repeated RoleAssignment role_assignments = 2;
+}
```

Comment:
I'm confused by the semantics of this RPC.  If my ListAssignments has:

```json
[
  {"joe", "admin"},
  {"fred", "editor"},
  {"sue", "admin"}
]
```

And I call UpdateRoleRequest with the following payload:

```json
[
  {"joe", "editor"},
  {"fred", "viewer"}
]
```

What's the expected outcome?

I'd expect something like:

```suggestion
message UpdateRoleRequest {
    // context is the context in which the role assignment is evaluated.
    Context context = 1;
    // subject is the account to change permissions for.
    // The account must already have permissions on the project
    subject string = 2;
    // All subject roles are _replaced_ with the following role assignments.  Must be non-empty,
    // use RemoveRole to remove permissions entirely from the project.
    repeated role string = 3;
}
```

Reply:
Repeating subject and project multiple times just seems like an invitation for funny business, so I'd just use direct fields here, unless this is going to be a PATCH method (you probably don't want that given our structures).

Reply:
oh, that's better, thanks! 👍 

You've switched the places of type and field name in the suggestion so I applied it separately